File: lib/chess_field.dart
import 'package:flutter/material.dart';
import 'chess_figure.dart';
import 'rammer_field.dart';

// Class name should be ChessField everywhere, not Field.
class ChessField {
  static const double fieldSize = 50.0;

  int x, y;
  List<int>? xyPosition;
  String? chessPosition;
  Color? color;
  RammerField? rammerField;
  bool changed;
  ChessFigure? figure;
  Color? marker;

  ChessField({
    required this.x,
    required this.y,
    this.xyPosition,
    this.chessPosition,
    this.color,
    this.rammerField,
    this.changed = false,
    this.figure,
    this.marker,
  });
}

File: doc/crazyChess prototype JS/rubikchess/chessField.js
export class ChessField {

    x;
    y;
    xyPosition;
    chessPosition;
    color;
    figure;
    rubikField;
    marker;

    constructor(x, y, xyPosition, chessPosition, color, figure, rubikField, marker) {
        this.id = `c_${x}${y}`;
        this.x = x;
        this.y = y;
        this.xyPosition = xyPosition;
        this.chessPosition = chessPosition;
        this.color = color;
        this.figure = figure;
        this.rubikField = rubikField;
        this.marker = marker;
    }

    toString() {
        return `xy:${this.xyPosition} color:${this.color} marker:${this.marker} 
                 |----figure:${this.rubikField} 
                 |----rubikField:${this.rubikField.toString()} `;
    }
}
